Change the translations to show a valid command that exits vim

alvarogarcia7 opened the pull request:
As stated in issues https://github.com/vim/vim/issues/1719 and https://github.com/vim/vim/issues/1720, plenty of new users of vim do not know how to exit vim.

Apparently, some of them type what they see on the screen `:quit<Enter>`, but this is not a valid vim command, as this produces a `E488: Trailing characters`.

The proposal is to change the message to `:quit|<Enter>`, which is a valid command (and exits vim). 

  * Those who type what they see, will exit vim
  * Those who get the idea of typing `:quit`, then `<Enter>` will also exit vim

alvarogarcia7 committed:
Change the translations to show a valid command that exits vim

brammool commented:
Alvaro Garcia wrote:

> As stated in issues https://github.com/vim/vim/issues/1719 and https://github.com/vim/vim/issues/1720, plenty of new users of vim do not know how to exit vim.
> 
> Apparently, some of them type what they see on the screen `:quit<Enter>`, but this is not a valid vim command, as this produces a `E488: Trailing characters`.
> 
> The proposal is to change the message to `:quit|<Enter>`, which is a valid command (and exits vim). 
> 
>   * Those who type what they see, will exit vim
>   * Those who get the idea of typing `:quit`, then `<Enter>` will also exit vim

People who don't get that <Enter> is one key still won't get out and
stare at the screen like a moron.

If even getting that <Enter> is a key is too difficult for some people,
how about this:

	Type  :quit  and press <Enter> to exit Vim

That's also easier to translate.  In case the user accidentally made
changes (hitting random keys) we could make it:

	Type  :qa!  and press <Enter> to abandon all changes and exit Vim

-- 
BLACK KNIGHT: The Black Knight always triumphs. Have at you!
   ARTHUR takes his last leg off.  The BLACK KNIGHT's body lands upright.
BLACK KNIGHT: All right, we'll call it a draw.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

alvarogarcia7 committed:
Revert "Change the translations to show a valid command that exits vim"

This reverts commit 7d888f0880bae0a4c59c8a64e1b2200447d4d583.

alvarogarcia7 commented:
Hello @brammool,

I will convert it to `Type  :quit  and press <Enter> to exit Vim`

How are the specific translations updated? Should I leave it empty, wrong, unmodified, so others can contribute translations in other languages?

Thanks,

alvarogarcia7 committed:
Refactor: change help message on Ctrl+c

Keep all translations, so they can be edited manually later

alvarogarcia7 committed:
Change Spanish translation

alvarogarcia7 committed:
Change Catalan translation

alvarogarcia7 committed:
Change French translation

alvarogarcia7 commented:
Hello @brammool,

I've changed the text as you suggested.

I've updated the `msgid`s in all the translation files. 

How can we translate the missing translation files?

Thanks in advance,

As a guide, I've translated:

  * ca.po
  * es.po
  * fr.po

But these translations are missing:

  * af.po
  * de.po
  * eo.po
  * fi.po
  * ga.po
  * it.po
  * ja.euc-jp.po
  * ja.po
  * ja.sjis.po
  * ko.UTF-8.po
  * ko.po
  * nb.po
  * nl.po
  * no.po
  * pl.UTF-8.po
  * pl.cp1250.po
  * pl.po
  * pt_BR.po
  * ru.cp1251.po
  * ru.po
  * sk.cp1250.po
  * sk.po
  * sv.po
  * uk.cp1251.po
  * uk.po
  * vi.po
  * zh_CN.UTF-8.po
  * zh_CN.cp936.po
  * zh_CN.po
  * zh_TW.UTF-8.po
  * zh_TW.po

 There are some translation files that do not need further action (e.g., en_GB), as they do not contain the original `msgid` in the first place.

brammool commented:
Alvaro Garcia wrote:

> Hello @brammool,
> 
> I've changed the text as you suggested.
> 
> I've updated the `msgid`s in all the translation files. 
> 
> How can we translate the missing translation files?

The maintainers of the translations need to do this.
Unfortunately some of them have not done anything recently, it's time to
poke them and perhaps find new translators.

> Thanks in advance,
> 
> As a guide, I've translated:
> 
>   * ca.po
>   * es.po
>   * fr.po
> 
> But these translations are missing:
> 
>   * af.po
>   * de.po
>   * eo.po
>   * fi.po
>   * ga.po
>   * it.po
>   * ja.euc-jp.po
>   * ja.po
>   * ja.sjis.po
>   * ko.UTF-8.po
>   * ko.po
>   * nb.po
>   * nl.po
>   * no.po
>   * pl.UTF-8.po
>   * pl.cp1250.po
>   * pl.po
>   * pt_BR.po
>   * ru.cp1251.po
>   * ru.po
>   * sk.cp1250.po
>   * sk.po
>   * sv.po
>   * uk.cp1251.po
>   * uk.po
>   * vi.po
>   * zh_CN.UTF-8.po
>   * zh_CN.cp936.po
>   * zh_CN.po
>   * zh_TW.UTF-8.po
>   * zh_TW.po
> 
>  There are some translation files that do not need further action (e.g., en_GB), as they do not contain the original `msgid` in the first place.

-- 
            |

Ceci n'est pas une pipe.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

dpelle commented on src/po/fr.po:
```diff
@@ -4338,7 +4338,7 @@ msgid "E663: At end of changelist"
 
 msgid "Type  :quit  and press <Enter> to exit Vim"
-msgstr "tapez  :q<Entr�e>  pour quitter Vim"
+msgstr "tapez  :q et alors la touche 'Entr�e'  pour quitter Vim"
```
The French translation is wrong, or not natural.  I'd rather say:

tapez  :q  puis la touche <Entrée> pour quitter Vim

I can update the French and Esperanto translations after this change is merged into the main line anyway, along with with other unrelated translation updates.

alvarogarcia7 replied:
Thanks, I've updated it

alvarogarcia7 committed:
Add French translation, as suggested by @dpelle

k-takata commented:
> How are the specific translations updated?

I think you should leave it unmodified. If you modify it, translators might not notice the change.

alvarogarcia7 commented:
Hello,

The coverage for this fell by a -0.009%. 

How can I fix this, so the PR goes through?

Thanks in advance,

k-takata commented:
> The coverage for this fell by a -0.009%.

It's probably because of compiler optimization or some flaky tests, not because of your change.
You don't need to care about it.

alvarogarcia7 commented:
Closing as this change made it to master already: https://github.com/vim/vim/commit/28a8193e3113f676f89fb6312b099d849df881d3
